heartbeat/mariadb-semi-sync

```diff
@@ -0,0 +1,1064 @@
+<longdesc lang="en">
+MariaDB replication user. This user is used for starting and stopping
+MariaDB replication, for setting and resetting the master host, and for
+setting and unsetting read-only mode. Because of that, this user must
+have SUPER, REPLICATION SLAVE, REPLICATION CLIENT, PROCESS and RELOAD
+privileges on all nodes within the cluster. Mandatory if you define a
+master-slave resource.
+</longdesc>
+<shortdesc lang="en">MariaDB replication user</shortdesc>
+<content type="string" default="${OCF_RESKEY_replication_user_default}" />
+</parameter>
+
+<parameter name="replication_passwd" unique="0" required="0">
+<longdesc lang="en">
+MariaDB replication password. Used for replication client and slave.
+Mandatory if you define a master-slave resource.
+</longdesc>
+<shortdesc lang="en">MariaDB replication user password</shortdesc>
+<content type="string" default="${OCF_RESKEY_replication_passwd_default}" />
+</parameter>
+
+<parameter name="replication_port" unique="0" required="0">
+<longdesc lang="en">
+The port on which the Master MariaDB instance is listening.
+</longdesc>
+<shortdesc lang="en">MariaDB replication port</shortdesc>
+<content type="string" default="${OCF_RESKEY_replication_port_default}" />
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start" timeout="120" />
+<action name="stop" timeout="120" />
+<action name="status" timeout="60" />
+<action name="monitor" depth="0" timeout="30" interval="20" />
+<action name="monitor" role="Master" depth="0" timeout="30" interval="10" />
+<action name="monitor" role="Slave" depth="0" timeout="30" interval="30" />
+<action name="promote" timeout="120" />
+<action name="demote" timeout="120" />
+<action name="notify" timeout="90" />
+<action name="validate-all" timeout="5" />
+<action name="meta-data" timeout="5" />
+</actions>
+</resource-agent>
+END
+}
+
+# Convenience functions
+
+greater_than_equal_long()
+{
+    # there are values we need to compare in this script
+    # that are too large for shell -gt to process
+    local true=$(echo "$1 > $2" | bc)
+    if [ "$true" -eq "1" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+greater_than_gtid()
+{
+    gtid1_transaction_id=$(echo $1 | cut -d - -f 3)
+    gtid2_transaction_id=$(echo $2 | cut -d - -f 3)
+
+    greater_than_equal_long $gtid1_transaction_id $gtid2_transaction_id
+    return $?
+}
+
+set_gtid() {
+    # Sets the GTID in CIB using attrd_updater for this node.
+    local gtid
+
+    gtid=$($MYSQL $MYSQL_OPTIONS_REPL \
+                  -s -N -e "show global variables like 'gtid_current_pos'" | cut -f 2)
+
+    # Ensure that we got somethine like a valid GTID
+    if ! echo $gtid | grep -q '-'; then
+        ocf_exit_reason "Unable to read GTID from MariaDB"
+        ocf_log err "Unable to read GTID from MariaDB"
+        return $OCF_ERR_GENERIC
+    fi
+
+    ${HA_SBIN_DIR}/attrd_updater -p -n ${OCF_RESOURCE_INSTANCE}-gtid -U $gtid
```

Comment:
Not a problem, just an FYI: using attrd_updater with -Q or -p limits the agent to clusters using pacemaker with corosync 2. Older cluster stacks use a "legacy attrd" that doesn't support those options. And of course this agent is pacemaker-specific (a few other software packages can use OCF agents). It might be worthwhile to document these limitations in the README and/or the meta-data.